roachprod: add virtual cluster support to the monitor

renatolabs opened the pull request:
This PR makes the roachprod monitor aware of multi-tenancy. This means
that the monitor is able to identify scenarios where multiple tenants
are running on the same VM, and monitor the lifecycle of tenant
processes just like it does for non multi-tenant deployments.

Under the covers, this is achieved by leveraging the
"ROACHPROD_VIRTUAL_CLUSTER" environment variable that is
set when starting cockroach. The monitor then looks for processes
that have that environment variable set, and parses it to identify the
tenant name and instance corresponding to that cockroach process.

This change should go unnoticed by all current monitor callers: tests
that monitor cockroach processes will continue to be notified of
events just like they did before. The only thing that changes is that,
if a test uses roachprod to start and stop tenants, these events will
be available through the monitor.

This PR also adds a `stop-tenant` command that uses the same
`ROACHPROD_TENANT` environment variable to allow the caller to
stop a specific tenant process.

Epic: none

Release note: None

Commit:
roachprod: add support for tags in `start-sql`

The help string for this subcommand already mentioned support for the
`tag` parameter; however, that was not implemented.

In this commit, we update the `startInstanceAsSeparateProcessCmd`
struct to accep the `--tag` flag and also apply a small fix to the
help string (app and storage nodes no longer need to be distinct).

Epic: none

Release note: None

Commit:
roachprod: add `stop-sql` subcommand

This adds a `roachprod stop-sql` command. It is similar to
`roachprod stop` in the sense that it takes a similar set of flags:
the signal to be sent to the processes, whether to wait for the
process to finish, and for how long. However, one crucial difference
is that `roachprod stop` stops *all* processes started by
roachprod (cockroach or otherwise), whereas `stop-sql` was designed
specifically for stopping tenant processes (SQL instances).

To achieve that, we set a `ROACHPROD_VIRTUAL_CLUSTER` environment
variable to the corresponding cockroach process when starting it. This
label is then used to find the correct process to stop when requested.

We also make use of the label to name the systemctl unit when starting
cockroach processes on remote clusters. This allows multiple tenants
to be co-located on the same VM.

Epic: none

Release note: None

Commit:
roachprod: add virtual cluster support to the monitor

This makes the roachprod monitor aware of virtual clusters. This means
that the monitor is able to identify scenarios where multiple
instances are running on the same VM, and monitor the lifecycle of sql
instance processes just like it does for non virtual cluster
deployments.

Under the covers, this is achieved by leveraging the
"ROACHPROD_VIRTUAL_CLUSTER" environment variable that is set when
starting cockroach. The monitor then looks for processes that have
that environment variable set, and parses it to identify the virtual
cluster name and instance corresponding to that cockroach process.

This change should go unnoticed by all current monitor callers: tests
that monitor cockroach processes will continue to be notified of
events just like they did before. The only thing that changes is that,
if a test uses roachprod to start and stop sql instances, these events
will be available through the monitor.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111064)
<!-- Reviewable:end -->

renatolabs commented:
Open for review. A few runs so far:

* `SELECT_PROBABILITY=0.1`, before rebasing against naming changes in #111113: https://teamcity.cockroachdb.com/viewLog.html?buildId=11941103&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel
* `SELECT_PROBABILITY=0.3`, after rebasing; still running, nothing unusual so far: https://teamcity.cockroachdb.com/viewLog.html?buildId=11955819&tab=queuedBuildOverviewTab

herkolategan reviewed (approved):
Nice work! Minor nits about tenant vs. virtual cluster naming and a few other things.
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-:-NfkovnY3S_8TiEq8OOU:b2jx739)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @renatolabs and @smg260)

___
*[`-- commits` line 38 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjpcBz8qAJRj1txjsR):*
Nit: Should this now be virtual cluster support?
___
*[`-- commits` line 43 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjpGJoAY3fkCYoc13B):*
Nit: Should this now be non virtual cluster deployments?
___
*[`pkg/cmd/roachprod/main.go` line 561 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjpqQ-19sw-CrifqN6:-NfjpqQ-19sw-CrifqN7:bik02km) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/cmd/roachprod/main.go#L561)):*
> ```Go
> 		}
> 
> 		// Always pick a random available port when starting tenants. We
> ```

Nit: virtual clusters.
___
*[`pkg/roachprod/multitenant.go` line 84 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjpzkt6gEbhVW1Lz9t:-Nfjpzkt6gEbhVW1Lz9u:b-14rgqc) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/multitenant.go#L84)):*
> ```Go
> // StopServiceForVirtualCluster stops SQL instance processes on the virtualCluster given.
> func StopServiceForVirtualCluster(
> 	ctx context.Context, l *logger.Logger, tenantCluster string, stopOpts StopOpts,
> ```

Nit: Do we want to update this to virtual cluster as well?
___
*[`pkg/roachprod/multitenant.go` line 99 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqADX9nUBkjuAAbsm:-NfjqADX9nUBkjuAAbsn:bd1bvuf) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/multitenant.go#L99)):*
> ```Go
> // tenant with ID given.
> //
> // TODO(herko): Allow users to pass in a tenant name.
> ```

Nit: Few mentions of tenant here again. As stated in the TODO we'll probably want to be able to pass virtual cluster name in the future directly once processes can start with a virtual cluster name instead of a virtual cluster id.
___
*[`pkg/roachprod/roachprod.go` line 716 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqQbI9diRnUmik7sL:-NfjqQbI9diRnUmik7sM:b-4w3sb0) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/roachprod.go#L716)):*
> ```Go
> 	MaxWait int
> 
> 	// Options that only apply to StopTenant
> ```

Nit: StopVirtualCluster?
___
*[`pkg/roachprod/install/cluster_synced.go` line 403 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqWWB31Pd2ysTgto_:-NfjqWWB31Pd2ysTgtoa:b-n01b43) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L403)):*
> ```Go
> // It sends a signal to all processes that have been started with ROACHPROD env
> // var and optionally waits until the processes stop. If the virtualClusterLabel
> // is not empty, then only tenant processes with a matching label are stopped.
> ```

Nit: sql processes.
___
*[`pkg/roachprod/install/cluster_synced.go` line 444 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjke1vAMc5L7AyLgoD:-Nfjke1vAMc5L7AyLgoE:b-694j6z) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L444)):*
> ```Go
> // for output/logging. If wait is true, the command will wait for the processes
> // to exit, up to maxWait seconds.
> // TODO(herko): This command does not support virtual clusters yet.
> ```

I suppose we can remove this comment now?
___
*[`pkg/roachprod/install/cluster_synced.go` line 647 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjs7jz2MVdHGha2hN_:-Nfjs7jz2MVdHGha2hNa:bkh0m3w) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L647)):*
> ```Go
> 	virtualClusterDesc := func(name string, instance int) string {
> 		if name == SystemInterfaceName {
> 			return "system tenant"
> ```

Nit: Wondering if we should call this system interface?
___
*[`pkg/roachprod/install/cluster_synced.go` line 746 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqkLh7KBiwtYUKFaA:-NfjqkLh7KBiwtYUKFaB:bik02km) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L746)):*
> ```Go
> 			)
> 			if err := errors.CombineErrors(err, result.Err); err != nil {
> 				err := errors.Wrap(err, "failed to list tenants")
> ```

Nit: virtual clusters.
___
*[`pkg/roachprod/install/cluster_synced.go` line 869 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjqpqg0Lwmf-55DuUM:-Nfjqpqg0Lwmf-55DuUN:b-bwc8gz) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L869)):*
> ```Go
> {{ end }}
> 
> # make sure all tenant monitors quit when this script exits. In
> ```

Nit: sql process / virtual cluster monitors?
___
*[`pkg/roachprod/install/cluster_synced.go` line 939 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjr3tT7J7tA0N0ulVl:-Nfjr3tT7J7tA0N0ulVm:b-wgb0ky) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L939)):*
> ```Go
> 						panic(fmt.Errorf("invalid output from monitor: %q", line))
> 					}
> 					// Tenant name and instance are the first fields of every
> ```

Nit: virtual cluster name.
___
*[`pkg/roachprod/install/cluster_synced.go` line 942 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjyi5a2SR7YHU9MErJ:-Nfjyi5a2SR7YHU9MErK:bqcx2iv) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L942)):*
> ```Go
> 					// event type.
> 					name, instanceStr := parts[0], parts[1]
> 					instance, _ := strconv.Atoi(instanceStr)
> ```

Nit: Should we maybe panic like above in the rare case that this string is not an integer?
___
*[`pkg/roachprod/install/cluster_synced.go` line 949 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjz6ca2e6PBLSStFsr:-Nfjz6ca2e6PBLSStFss:b-whlnyt) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L949)):*
> ```Go
> 						}})
> 					case runningMsg:
> 						pid := parts[3]
> ```

Nit: Should we add a `len < 4` panic here as above?
___
*[`pkg/roachprod/install/cluster_synced.go` line 954 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjzVEPAy-7n6AZ_o41:-NfjzVEPAy-7n6AZ_o42:b-whlnyt) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L954)):*
> ```Go
> 						}})
> 					case deadMsg:
> 						exitCode := parts[3]
> ```

Nit: Should we add a `len < 4` panic here as above?
___
*[`pkg/roachprod/install/cockroach.go` line 602 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjr9LD92tX96DfXSrG:-Nfjr9LE2SEipK5oH7TN:b-fll170) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach.go#L602)):*
> ```Go
> }
> 
> // TenantInfoFromLabel takes as parameter a tenant label produced with
> ```

Nit: VirtualClusterInfoFromLabel ... virtual cluster label.
___
*[`pkg/roachprod/install/cockroach_test.go` line 27 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsMKyAFbY_4KY-80m:-NfjsMKzEH0aKVZHhfPp:bxwixbv) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L27)):*
> ```Go
> 	}{
> 		{
> 			name:               "empty tenant name",
> ```

Nit: empty virtual cluster name.
___
*[`pkg/roachprod/install/cockroach_test.go` line 32 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsFMLAJ0cQmF0mtPV:-NfjsFMLAJ0cQmF0mtPW:bpupdli) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L32)):*
> ```Go
> 		},
> 		{
> 			name:               "system tenant name",
> ```

Nit: System interface name?
___
*[`pkg/roachprod/install/cockroach_test.go` line 37 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsRPp5hn2cTvF-m-p:-NfjsRPp5hn2cTvF-m-q:b-8z3naj) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L37)):*
> ```Go
> 		},
> 		{
> 			name:               "simple app tenant name",
> ```

Nit: simple app virtual cluster name.
___
*[`pkg/roachprod/install/cockroach_test.go` line 43 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsUiVCe_3va7DiXf-:-NfjsUiVCe_3va7DiXf0:b-d2t74p) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L43)):*
> ```Go
> 		},
> 		{
> 			name:               "tenant name with hyphens",
> ```

Nit: virtual cluster name with hyphens.
___
*[`pkg/roachprod/install/cockroach_test.go` line 58 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjrN_c8g89jwzcTG02:-NfjrN_c8g89jwzcTG03:b-i3f1a6) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L58)):*
> ```Go
> 			require.NoError(t, err)
> 
> 			expectedTenantName := tc.virtualClusterName
> ```

Nit: expectedVirtualClusterName.


<!-- Sent from Reviewable.io -->

renatolabs reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111064)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @herkolategan and @smg260)

___
*[`-- commits` line 38 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjpcBz8qAJRj1txjsR):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Should this now be virtual cluster support?
</blockquote></details>

Done.
___
*[`-- commits` line 43 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjpGJoAY3fkCYoc13B):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Should this now be non virtual cluster deployments?
</blockquote></details>

Done.
___
*[`pkg/cmd/roachprod/main.go` line 561 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjpqQ-19sw-CrifqN6:-Nflu28NBYR9q8dMw7Wm:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/cmd/roachprod/main.go#L561)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: virtual clusters.
</blockquote></details>

Done.
___
*[`pkg/roachprod/multitenant.go` line 84 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjpzkt6gEbhVW1Lz9t:-Nflu3U_CcxHQ4JWbvz_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/multitenant.go#L84)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Do we want to update this to virtual cluster as well?
</blockquote></details>

Done.
___
*[`pkg/roachprod/multitenant.go` line 99 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqADX9nUBkjuAAbsm:-Nflu4qo-d1cNW9BDTt2:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/multitenant.go#L99)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Few mentions of tenant here again. As stated in the TODO we'll probably want to be able to pass virtual cluster name in the future directly once processes can start with a virtual cluster name instead of a virtual cluster id.
</blockquote></details>

Done.
___
*[`pkg/roachprod/roachprod.go` line 716 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqQbI9diRnUmik7sL:-Nflu5hx3aOZSmevVn2z:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/roachprod.go#L716)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: StopVirtualCluster?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 403 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqWWB31Pd2ysTgto_:-Nflu6Zk4O9L9IjHbJDP:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L403)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: sql processes.
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 444 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjke1vAMc5L7AyLgoD:-Nflu7DGEYbkZYiaT2Gc:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L444)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

I suppose we can remove this comment now?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 647 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjs7jz2MVdHGha2hN_:-Nflu7wA1P-XiF_h25_R:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L647)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Wondering if we should call this system interface?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 746 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjqkLh7KBiwtYUKFaA:-Nflu8dY9VdQGQz-Oor_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L746)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: virtual clusters.
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 869 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjqpqg0Lwmf-55DuUM:-Nflu9Ci5NYHBYBNdaYx:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L869)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: sql process / virtual cluster monitors?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 939 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjr3tT7J7tA0N0ulVl:-Nflu9t-13_ZxYgcT8KL:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L939)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: virtual cluster name.
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 942 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjyi5a2SR7YHU9MErJ:-NfluAei-iDOI5hAdcdx:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L942)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Should we maybe panic like above in the rare case that this string is not an integer?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 949 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjz6ca2e6PBLSStFsr:-NfluBiu5N0_VYPQhlv2:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L949)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Should we add a `len < 4` panic here as above?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cluster_synced.go` line 954 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjzVEPAy-7n6AZ_o41:-NfluCLpCv2p7JgoZ-ZY:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cluster_synced.go#L954)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: Should we add a `len < 4` panic here as above?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cockroach.go` line 602 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfjr9LD92tX96DfXSrG:-NfluDUy2N5dHu9BeUQa:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach.go#L602)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: VirtualClusterInfoFromLabel ... virtual cluster label.
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cockroach_test.go` line 27 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsMKyAFbY_4KY-80m:-NfluEJ10VoCcGAUnhee:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L27)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: empty virtual cluster name.
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cockroach_test.go` line 32 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsFMLAJ0cQmF0mtPV:-NfluEml9cVYtFmA83d5:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L32)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: System interface name?
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cockroach_test.go` line 37 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsRPp5hn2cTvF-m-p:-NfluFm74iQQ8C5Knhtr:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L37)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: simple app virtual cluster name.
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cockroach_test.go` line 43 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjsUiVCe_3va7DiXf-:-NfluGGrEJg9DZpBjAou:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L43)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: virtual cluster name with hyphens.
</blockquote></details>

Done.
___
*[`pkg/roachprod/install/cockroach_test.go` line 58 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-NfjrN_c8g89jwzcTG02:-NfluGsAANexvzLy9oCf:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9cbeda575cfe3bfa3d6cbfcecbfa46a5266c04fe/pkg/roachprod/install/cockroach_test.go#L58)):*
<details><summary><i>Previously, herkolategan (Herko Lategan) wrote…</i></summary><blockquote>

Nit: expectedVirtualClusterName.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

renatolabs commented:
Rebased and addressed comments. Running another subset of tests just to be sure. If everything is well, plan to merge this tomorrow:

https://teamcity.cockroachdb.com/viewQueued.html?itemId=12021678&tab=queuedBuildOverviewTab

renatolabs commented:
Build finished without any unexpected failures. Merging. TFTR!

bors r=herkolategan

renatolabs commented:
bors r-

srosenberg reviewed (commented):
> Reviewed 2 of 14 files at r1, 1 of 9 files at r2, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111064)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @herkolategan, @renatolabs, and @smg260)

___
*[`pkg/cmd/roachprod/main.go` line 729 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfrp-hY19KBJ5mHmQ3f:-Nfrp-hY19KBJ5mHmQ3g:b8vf5nd) ([raw file](https://github.com/cockroachdb/cockroach/blob/86ead7dfaf15decc784e66b16988080b66e0a695/pkg/cmd/roachprod/main.go#L729)):*
> ```Go
> 	Use:   "signal <cluster> <signal>",
> 	Short: "send signal to cluster",
> 	Long:  "Send a POSIX signal, specified by its integer code, to every roachprod process in a cluster.",
> ```

Nit: I had to spend a few minutes tracking down what's meant by "every roachprod process". I realize this is external documentation, but it might be worth including a reference to `roachprodEnvValue` for those who seek more context. There is actually `opts.includeRoachprodEnvVars` which may spawn processes without `export ROACHPROD=`, but it's rarely (if ever) used.


<!-- Sent from Reviewable.io -->

renatolabs reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111064)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @herkolategan, @smg260, and @srosenberg)

___
*[`pkg/cmd/roachprod/main.go` line 729 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/111064#-Nfrp-hY19KBJ5mHmQ3f:-NfvjUHgDKIfhyorhEki:bhbhkvl) ([raw file](https://github.com/cockroachdb/cockroach/blob/86ead7dfaf15decc784e66b16988080b66e0a695/pkg/cmd/roachprod/main.go#L729)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Nit: I had to spend a few minutes tracking down what's meant by "every roachprod process". I realize this is external documentation, but it might be worth including a reference to `roachprodEnvValue` for those who seek more context. There is actually `opts.includeRoachprodEnvVars` which may spawn processes without `export ROACHPROD=`, but it's rarely (if ever) used.
</blockquote></details>

Rephrased this a bit to say "every process started by roachprod" which is close enough to the truth to be useful without needing to be extra long to be completely accurate.


<!-- Sent from Reviewable.io -->

renatolabs commented:
Another run with `SELECT_PROBABILITY=0.3`: https://teamcity.cockroachdb.com/viewLog.html?buildId=12054655&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel&tab=buildResultsDiv&branch_Cockroach_Nightlies=111064.

Also manually triggered the `Local roachtest` build in TC to make sure there are no obvious flakes in local runs. Nothing found.

renatolabs commented:
TFTR!

bors r=herkolategan
